testnet/stacks-node/src/tests/signer/v0.rs

```diff
@@ -579,6 +579,217 @@ fn miner_gather_signatures() {
+
+    // Make all but 1 signers ignore. We need 1 signer to view it so that they save
+    // it in their SignerDB. If a signer has never seen a block, they won't
+    // handle other signer's block responses, which is what causes the signer to
+    // eventually broadcast the block.
+    let ignoring_signers: Vec<_> = all_signers.iter().cloned().take(num_signers - 1).collect();
+    info!(
+        "----- Ignoring block proposals from {} of {} signers -----",
+        ignoring_signers.len(),
+        all_signers.len()
+    );
+    TEST_IGNORE_ALL_BLOCK_PROPOSALS
+        .lock()
+        .unwrap()
+        .replace(ignoring_signers);
+
+    let proposals_before = signer_test
+        .running_nodes
+        .nakamoto_blocks_proposed
+        .load(Ordering::SeqCst);
+
+    // submit a tx so that the miner will mine an extra block
+    let sender_nonce = 0;
+    let transfer_tx = make_stacks_transfer(
+        &sender_sk,
+        sender_nonce,
+        send_fee,
+        signer_test.running_nodes.conf.burnchain.chain_id,
+        &recipient_addr.into(),
+        send_amt,
+    );
+    submit_tx(&http_origin, &transfer_tx);
+
+    info!("Submitted transfer tx and waiting for block proposal");
+    wait_for(30, || {
+        Ok(signer_test
+            .running_nodes
+            .nakamoto_blocks_proposed
+            .load(Ordering::SeqCst)
+            > proposals_before)
+    })
+    .expect("Timed out waiting for block proposal");
+
+    let block_proposal = test_observer::get_stackerdb_chunks()
+        .into_iter()
+        .flat_map(|chunk| chunk.modified_slots)
+        .filter_map(|chunk| SignerMessage::consensus_deserialize(&mut chunk.data.as_slice()).ok())
+        .filter_map(|message| {
+            let SignerMessage::BlockProposal(proposal) = message else {
+                return None;
+            };
+            let has_transfer = proposal.block.txs.iter().any(|tx| {
+                let TransactionPayload::TokenTransfer(recipient_principal, ..) = tx.clone().payload
+                else {
+                    return false;
+                };
+                recipient_principal == recipient_addr.into()
+            });
+            if has_transfer {
+                Some(proposal)
+            } else {
+                None
+            }
+        })
+        .last()
+        .expect("Expected a block proposal to be found");
+
+    let block = block_proposal.block;
+
+    // First propose a block to the signers that does not have the correct consensus hash or BitVec. This should be rejected BEFORE
+    // the block is submitted to the node for validation.
+    let block_signer_signature_hash_1 = block.header.signer_signature_hash();
+    info!(
+        "------ Proposing Block ------";
+        "signer_sig_hash" => %block_signer_signature_hash_1,
+    );
+
+    signer_test.propose_block(block.clone(), short_timeout);
+    info!("------ Mining BTC Block ------");
+    next_block_and_wait(
```

Comment:
Should we record the current `nakamoto_blocks_proposed` here, so that we can make sure no other blocks were proposed before the target block?

Reply:
I ended up removing the `propose_block` line, as it wasn't actually necessary - the proposal is already in StackerDB. If anything, maybe we want to verify that this block isn't already actually confirmed before the next tenure?

Reply:
Yeah, that would make sense, but what I was suggesting is to check that the new miner does not submit a proposal before this block is approved and broadcasted.

Reply:
Ah yeah, makes sense. I've added a check during the new tenure to make sure the new miner hasn't proposed any blocks.